src/components/AccountSelector/AccountSelectButton/AccountSelectorButton.tsx

```diff
@@ -0,0 +1,56 @@
+import { ComponentPropsWithoutRef } from "react";
+
+export interface AccountSelectorProps extends ComponentPropsWithoutRef<"div"> {
+  theme?: string;
+  profileimage: React.ReactNode;
+  accountname: string;
+  svg: React.ReactNode;
+  accountbalance: string;
+  children?: React.ReactNode;
+}
+
+export function AccountSelectorButton(props: AccountSelectorProps) {
+  // Specific classes for the account selector button
+  const accountSelectorClass = "flex flex-row items-center justify-between";
+
+  const hoverClass = `hover:bg-info`;
+  const activeClass = `active:bg-c-pressed`;
+
+  // Generic classes
+  const genericFlex = "flex justify-center items-center";
+
+  const tokenDimensionClass = `h-6 w-6`;
+
+  return (
+    <div
+      className={` bg-neutral text-primary h-[56px] w-full p-3 rounded-lg mas-menu-active cursor-pointer 
+                      ${accountSelectorClass} 
+                      ${hoverClass} 
+                      ${activeClass}`}
+      {...props}
+      data-testid="account-selector-button"
+    >
+      <div className={`${genericFlex}`}>
+        <div
+          className={`bg-primary h-8 w-8 rounded-full mr-2
+                           ${genericFlex}`}
+        >
+          {props.profileimage}
```

Comment:
Please, unstructure props in the component beginning and use the attribute only here. It's easy to understand all params we are working.

```let { attr, attr-2, ... } = props```

Reply:
Updated !